### PR TITLE
NXP-24969: aws transcribe for videos

### PR DESCRIPTION
**Pull request by andreinechaev**

### CODE REVIEW

File: addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/transcribe/TranscribeWork.java [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+            throw new NuxeoException("No supported audio conversion found. Transcribe requires " + DEFAULT_CONVERSION);
+        }
+
+        if (!doc.hasFacet(TRANSCRIBE_FACET)) {
+            doc.addFacet(TRANSCRIBE_FACET);
+        }
+
+        TranscribeService service = Framework.getService(TranscribeService.class);
+        CompletableFuture<List<Alternative>> future = service.transcribe(wav.getBlob(), DEFAULT_LANG_CODE);
+        List<Alternative> result = future.join();
```
Comment (vpasquier) [inline addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/transcribe/TranscribeWork.java]: is there any thing to take into account if the transcription is taking a long long time (for a movie of big size)? I don't see any transaction or session timeout issue here but did we try to do some tests with big assets to make sure?

File: addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/transcribe/TranscribeComponent.java [hunk 2]
```diff
@@ -0,0 +1,220 @@
...
+import software.amazon.awssdk.services.transcribestreaming.TranscribeStreamingAsyncClient;
+import software.amazon.awssdk.services.transcribestreaming.model.Alternative;
+import software.amazon.awssdk.services.transcribestreaming.model.Item;
+import software.amazon.awssdk.services.transcribestreaming.model.LanguageCode;
+import software.amazon.awssdk.services.transcribestreaming.model.MediaEncoding;
+import software.amazon.awssdk.services.transcribestreaming.model.StartStreamTranscriptionRequest;
+import software.amazon.awssdk.services.transcribestreaming.model.StartStreamTranscriptionResponseHandler;
+import software.amazon.awssdk.services.transcribestreaming.model.TranscriptEvent;
+
+public class TranscribeComponent extends DefaultComponent implements TranscribeService {
```
[the remark this answers is not in the document]
Comment (andreinechaev) [reply addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/transcribe/TranscribeComponent.java]: maybe you're right. I thought that we might have extensions for supporting different languages.

File: addons/nuxeo-ai-aws-core/src/main/resources/OSGI-INF/listener-contrib.xml [hunk 3]
```diff
@@ -37,5 +37,13 @@
       <event>asyncRekognitionCelebritiesSuccess</event>
       <event>asyncRekognitionCelebritiesFailure</event>
     </listener>
+
+    <listener name="audioConversionReadyForTranscribe"
+              async="true"
+              postCommit="true"
+              priority="999"
+              class="org.nuxeo.ai.transcribe.AudioConversionReady">
+      <event>videoConversionsDone</event>
```
[the remark this answers is not in the document]
Comment (andreinechaev) [reply addons/nuxeo-ai-aws-core/src/main/resources/OSGI-INF/listener-contrib.xml]: it's a default Nuxeo event fired by `VideoConversionWork`
I added `WAV` conversion in NXP-27915

File: addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/transcribe/TranscribeWork.java (hunk 1 above)
Comment (andreinechaev) [reply addons/nuxeo-ai-aws-core/src/main/java/org/nuxeo/ai/transcribe/TranscribeWork.java]: good catch. need to revisit

**Pull request merged**